db/external_sst_file_ingestion_job.cc

```diff
@@ -622,7 +622,10 @@ Status ExternalSstFileIngestionJob::GetIngestedFileInfo(
     if (key.sequence != 0) {
       return Status::Corruption("External file has non zero sequence number");
     }
-    file_to_ingest->smallest_internal_key.SetFrom(key);
+
+    SequenceNumber assigned_seqno = versions_->LastSequence();
```

Comment:
Maybe I am missing some context from the original bug, but I think we may later call `AssignLevelAndSeqnoForIngestedFile(...)` to compute `assigned_seqno` which does not have to be the current latest seqno in the db.

Reply:
just did a bit of refactoring to include that part.